OMEdit/OMEditGUI/Editors/ModelicaEditor.cpp

```diff
@@ -183,6 +198,72 @@ void ModelicaEditor::getCompletionSymbols(QString word, QList<CompleterItem> &cl
   }
 }
 
+void ModelicaEditor::getCompletionAnnotations(const QStringList &stack, QList<CompleterItem> &annotations)
+{
+  QString key = stack.join(".");
```

Comment:
What is the purpose of having a `QStringList` here. I guess you are always expecting a scalar value.

Reply:
Now, it is always **converted** to scalar `QString` value, but it is at least supposed to be sequence like `annotation(a(b(cd<cursor is here>=1)))) --> ["a","b"]`

Reply:
We should build a tree structure of annotations then.
Well anything that works is fine. The only thing is it should be good enough to extend it with future annotations. Right now we have so many annotations and I prefer just one place to specify them once.

Reply:
Maybe collect them in some help package in their `record`-like form? Then we could specify description strings and other metadata, not only names.

Reply:
Then we have to do this in the OMCompiler. I prefer some solution in OMEdit.

Reply:
Just to clarify: I don't propose to implement some auto-generated package in OMCompiler, I just propose to make some Modelica **library** just like `ModelicaReference.Annotations` that would be made just of records closely resembling actual annotations. Unfortunately, current documentation looks human-, but not very machine-readable.

This would make annotation descriptions:
* easy to edit "in their textbook form" :)
* inherently hierarchical, no need to reimplement the wheel in C++
* they would have data types and descriptions
* they would be edited by `OMCompiler` team :)

Reply:
Meanwhile, it would be interesting to automatically fetch descriptions for completer keywords from `ModelicaReference`.